src/Microsoft.OData.Core/Json/ODataJsonPropertyAndValueDeserializer.cs

```diff
@@ -217,7 +218,8 @@ internal static IEdmTypeReference ResolveUntypedType(
                 IEdmTypeReference payloadTypeReference,
                 Func<object, string, IEdmTypeReference> primitiveTypeResolver,
                 bool readUntypedAsString,
-                bool generateTypeIfMissing)
+                bool generateTypeIfMissing,
+                bool readUntypedNumericAsDecimal = true)

```

Comment:
In ODL 9, should we have this default to `true`, of have it default to `false` while allowing folks to toggle back legacy behaviour via the compatibility flag?